app/models/user.rb

```diff
@@ -163,31 +163,43 @@ def course_student?
   end
 
   def nonvisitor?(course)
-    role(course) != CoursesUsers::Roles::VISITOR_ROLE
+    return true if admin?
+    roles = course_roles(course)
+    roles.size > 1 || (roles.size == 1 && roles.first != CoursesUsers::Roles::VISITOR_ROLE)
```

Comment:
The visitor role does not exist as a CoursesUsers record, so it should never be among the roles returned by `course_roles`

Reply:
You are right, @ragesoss . I guess a better approach will be to check that the `course_roles` array is not empty, right?

Reply:
yeah. course_role().empty? should work.